modules/transports/core/vfs/src/main/java/org/apache/synapse/transport/vfs/VFSTransportListener.java

```diff
@@ -881,6 +890,12 @@ private void processFile(PollTableEntry entry, FileObject file) throws AxisFault
                     null, //* SOAP Action - not applicable *//
                     contentType
                 );
+                //Check if the sender was successful or not
+                String errorCode = (String) transportHeaders.get(VFSConstants.ERROR_CODE);
```

Comment:
Can we check a synapse level parameter rather than transport scope, if this is transport, this will always a custom property that we have to set in fault sequence

Reply:
Issue here is Existing code creates axis2 context in the VFSTransportListner and injects it to the axis.engine within AbstractTransportListner
And from that method what we receive is also an axis2 message context so we don't have the access to synapse context in this transport layer